pandas/core/internals/array_manager.py

```diff
@@ -270,15 +270,30 @@ def grouped_reduce(self: T, func: Callable, ignore_failures: bool = False) -> T:
         -------
         ArrayManager
         """
-        # TODO ignore_failures
-        result_arrays = [func(arr) for arr in self.arrays]
+        result_arrays: List[np.ndarray] = []
```

Comment:
this looks a whole lot like reduce right?

Reply:
Yes, it's quite similar in logic (it are also both reduce operations, so not that unsurprising), but IMO they are different enough that trying to share anything will only make it more complex (return value is different, they need to process the result inside the loop differently, etc)

It might be possible to change the return value of `reduce` to make this easier, but that's a bigger change, so if we want that, it's for a separate PR